Assets/Plugins/Editor/Uplift/UpdateTool/Updater.cs

```diff
@@ -0,0 +1,131 @@
+using UnityEngine;
+using UnityEditor;
+using System.IO;
+using System.Collections;
+using System.Collections.Generic;
+using Uplift.Common;
+using Uplift.Windows;
+using System.Net;
+using System.Linq;
+using System;
+using System.Globalization;
+
+namespace Uplift.Updating
+{
+    public class Updater : MonoBehaviour
+    {
+        private static readonly string dateFormat = @"MM\/dd\/yyyy HH:mm";
+        private static readonly CultureInfo provider = CultureInfo.InvariantCulture;
+        private static readonly string lastUpdateCheckKey = "UpliftLastUpdateCheck";
+        private static IEnumerator updateCoroutine;
+
+        public static void UpdateUplift(string url)
+        {
+            string unitypackageName = url.Split('/').Last();
+            DirectoryInfo assetsPath = new DirectoryInfo(Application.dataPath);
+            string destination = Path.Combine(assetsPath.Parent.FullName, unitypackageName);
+            
+            ServicePointManager.ServerCertificateValidationCallback = delegate { return true; };
+            using (var client = new WebClient())
+            {
+                client.DownloadFile(url, destination);
+            }
+
+            AssetDatabase.ImportPackage(destination, false);
+        }
+
+        public static void CheckForUpdate(bool forceCheck = false)
+        {
+            if(forceCheck || ShouldCheck())
+            {
+                EditorApplication.update += EditorUpdate;
+    			updateCoroutine = CheckForUpdateRoutine();
+            }
+        }
+        
+        private static void EditorUpdate()
+        {
+            updateCoroutine.MoveNext();
+        }
+
+        private static bool ShouldCheck()
+        {
+            string lastCheck = EditorPrefs.GetString(lastUpdateCheckKey);
```

Comment:
Would that prevent update in a different project?

Reply:
I'm not sure. It could actually. I'll check this.

Reply:
It looks like the preferences are project specific!

Reply:
Cool.